ouroboros-consensus-byron/src/Ouroboros/Consensus/Byron/Node/Serialisation.hs

```diff
@@ -154,3 +242,63 @@ instance SerialiseNodeToClient ByronBlock (Some (Query ByronBlock)) where
 instance SerialiseResult ByronBlock (Query ByronBlock) where
   encodeResult _ _ = encodeByronResult
   decodeResult _ _ = decodeByronResult
+
+{-------------------------------------------------------------------------------
+  Nested contents
+-------------------------------------------------------------------------------}
+
+instance ReconstructNestedType Header ByronBlock where
+  reconstructPrefixLen _ = 0
+  reconstructNestedType _proxy isEBB size _bs =
+      case isEBB of
+        IsEBB    -> SomeBlock $ NestedType (BoundaryByronType size)
+        IsNotEBB -> SomeBlock $ NestedType (RegularByronType  size)
+
+instance EncodeDiskDep (NestedType Header) ByronBlock where
+  encodeDiskDepIx ByronCodecConfig{..} (SomeBlock (NestedType typ)) = mconcat [
+        CBOR.encodeListLen 2
+      , case typ of
+          RegularByronType size -> mconcat [
+              CBOR.encodeWord8 0
+            , CBOR.encodeWord32 size
+            ]
+          BoundaryByronType size -> mconcat [
+              CBOR.encodeWord8 1
+            , CBOR.encodeWord32 size
+            ]
+      ]
+
+  encodeDiskDep ByronCodecConfig{..} (NestedType typ) h =
+      case typ of
+        RegularByronType _size ->
+          encodeByronRegularHeader h
+        BoundaryByronType _size ->
+          -- We don't encode the 'SlotNo'
```

Comment:
Comment: this is important, because this should be compatible with the raw bytes on disk.